portfolio/src/main/webapp/scripts/snake/Snake.js

```diff
@@ -0,0 +1,130 @@
+     * @param {number} xConstraint1 First x-coordinate constraint
+     * @param {number} xConstraint2 Second x-coordinate constraint
+     * @param {number} yConstraint1 First y-coordinate constraint
+     * @param {number} yConstraint2 Second y-coordinate constraint
+     * @param {number} gridScale Scale of game grid that food will be rendered on
+     * @param {number} red RGB code (0 - 255)
+     * @param {number} green RGB code (0 - 255)
+     * @param {number} blue RGB code (0 - 255)
+     */
+    constructor({x, y, xConstraint1, xConstraint2, yConstraint1, yConstraint2, gridScale, red, green, blue}) {
+        // The snake is not moving upon initialization
+        // @private {number} Snake speed in the x-direction
+        this.xSpeed = 0;
+        // @private {number} Snake speed in the y-direction
+        this.ySpeed = 0;
+
+        // @private {Array<Vector>} Array of vectors representing each tail element of the snake
+        this.tail = [];
+
+        // @private {number} Total number of tail pieces
+        this.total = 0;
+
+        this.x = x;
+        this.y = y;
+
+        this.xConstraint1 = xConstraint1;
+        this.xConstraint2 = xConstraint2;
+        this.yConstraint1 = yConstraint1;
+        this.yConstraint2 = yConstraint2;
+
+        this.gridScale = gridScale;
+
+        this.red = red;
+        this.green = green;
+        this.blue = blue;
+    }
+
+    /**
+     * Sets speed and direction of snake
+     */
+    dir(xSpeed, ySpeed) {
+        this.xSpeed = xSpeed;
+        this.ySpeed = ySpeed;
+    }
+
+    /**
+     * Renders each tail element and head of the snake as rectangles on a p5 canvas.
+     */
+    show() {
+        // Show all blocks in the tail
+        for (let i = 0; i < this.tail.length; i++) {
+            fill(this.red, this.green, this.blue);
+            rect(this.tail[i].x, this.tail[i].y, this.gridScale, this.gridScale);
+        }
+        // Show the head
+        fill(this.red, this.green, this.blue);
+        rect(this.x, this.y, this.gridScale, this.gridScale);
+    }
+
+    /**
+     * Determines if food at a certain x and y-coordinate are close enough to the
+     * snake to be "eaten".
+     * @return {boolean} True if close enough to be eaten, else false.
+     */
+    eat(foodX, foodY) {
+        const distanceFromFood = dist(this.x, this.y, foodX, foodY);
+        if (distanceFromFood < 10) {
+            this.total++;
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Determine if Snake has died by checking whether it has hit bounds or hit its
+     * own tail.
+     * @return {boolean} True if dead, else false.
+     */
+    death() { return this.checkIfHitTail() }
+
+    /**
+     * Determines if Snake has hit any element of its tail.
+   * When the Snake hits a boundary, the tail collapses so 
+   * the Snake hits its tail whenever it hits a boundary.
+     * @return {boolean} True if hit tail, else false.
+     */
+    checkIfHitTail() {
```

Comment:
I think it would make sense for this to just be the death() function instead of having one that is simply a wrapper for this if this is not called anywhere else, which it doesn't seem to be.  Other than saving needless additional methods, I think it's a little confusing that the death() function seems to indicate it checks two conditions when really it only checks one (because they mean the same thing) which can be more clearly explained here.